browser/resources/settings/brave_overrides/site_settings_page.js

```diff
@@ -14,15 +16,17 @@ const CONTENT_ADVANCED_REMOVE_IDS = [
   ContentSettingsTypes.ADS,
 ]
 
-RegisterPolymerComponentBehaviors({
-  'settings-site-settings-page': [{
-    registered: function() {
-      if (!this.properties || !this.properties.lists_ || !this.properties.lists_.value) {
+RegisterPolymerComponentReplacement(
+  'settings-site-settings-page',
+  class BraveComponent extends SettingsSiteSettingsPageElement {
```

Comment:
It's taken a while to digest this all; but the changes you made make complete sense. Now that you have an example worked out, it's "not too terrible". But some pretty intense work you had to do to get here!